### ISSUE TITLE
Stopping after a few minutes

### ISSUE THREAD

**Issue by cdcaish** [ZoeyVid/NPMplus#1013]
I'm hoping you can maybe point me in the direction of additional logs or help with troubleshooting.  
I'm on version 2.11.3+dbb0883 hosted on TrueNAS Scale 24.04.2.  443/80 TCP/UDP are forwarded from my router to NPMplus.  I'm also using duckdns.

I have 1 proxy host configured and it will work fine outside of my LAN for about 5 minutes and then it is no longer reachable.  If I stop/start NPMplus it'll work again for another 5 minutes before becoming unreachable again.  When it becomes unreachable outside I am still able to reach it inside my LAN.  NPMplus webui is still responsive though it seems like NPMplus stops responding to the proxy requests.

Any help would be greatly appreciated!

**Comment by Zoey2936** [comment]
can you send me your logs?

**Comment by cdcaish** [comment]
Thanks for the quick response!  Let me know if there is a different log you are looking for. 

```
2024-08-13 16:04:43.307984-04:00'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/README.md' -> '/data/etc/modsecurity/crs-plugins/README.md'
2024-08-13 16:04:43.308748-04:00'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-after.conf' -> '/data/etc/modsecurity/crs-plugins/empty-after.conf'
2024-08-13 16:04:43.308779-04:00'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-before.conf' -> '/data/etc/modsecurity/crs-plugins/empty-before.conf'
2024-08-13 16:04:43.308804-04:00'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-config.conf' -> '/data/etc/modsecurity/crs-plugins/empty-config.conf'
2024-08-13 16:04:43.309398-04:00no DEFAULT_CERT_ID set, using dummycerts.
2024-08-13 16:04:44.457713-04:00Working on file: /data/nginx/custom/events.conf
2024-08-13 16:04:44.461300-04:00Working on file: /data/nginx/custom/http.conf
2024-08-13 16:04:44.461571-04:00Working on file: /data/nginx/custom/http_top.conf
2024-08-13 16:04:44.461789-04:00Working on file: /data/nginx/custom/root_top.conf
2024-08-13 16:04:44.462035-04:00Working on file: /data/nginx/custom/server_dead.conf
2024-08-13 16:04:44.462407-04:00Working on file: /data/nginx/custom/server_proxy.conf
2024-08-13 16:04:44.462613-04:00Working on file: /data/nginx/custom/server_redirect.conf
2024-08-13 16:04:44.463129-04:00Working on file: /data/nginx/custom/server_stream.conf
2024-08-13 16:04:44.463357-04:00Working on file: /data/nginx/custom/server_stream_tcp.conf
2024-08-13 16:04:44.463722-04:00Working on file: /data/nginx/custom/server_stream_udp.conf
2024-08-13 16:04:44.463916-04:00Working on file: /data/nginx/custom/stream.conf
2024-08-13 16:04:44.464603-04:00Working on file: /data/nginx/custom/stream_top.conf
2024-08-13 16:04:44.464831-04:00Working on file: /data/nginx/default.conf
2024-08-13 16:04:44.467887-04:00Working on file: /data/nginx/ip_ranges.conf
2024-08-13 16:04:44.469103-04:00Working on file: /data/nginx/proxy_host/2.conf
2024-08-13 16:04:44.469795-04:00Success.
2024-08-13 16:04:44.692411-04:002024-08-13T16:04:44.692411112-04:00
2024-08-13 16:04:44.692488-04:00-------------------------------------
2024-08-13 16:04:44.692509-04:00_ _  ___  __ __       _
2024-08-13 16:04:44.692542-04:00| \ || . \|  \  \ ___ | | _ _  ___
2024-08-13 16:04:44.692567-04:00|   ||  _/|     || . \| || | |[_-[
2024-08-13 16:04:44.692621-04:00|_\_||_|  |_|_|_||  _/|_| \__|/__/
2024-08-13 16:04:44.692642-04:00|_|
2024-08-13 16:04:44.692656-04:00-------------------------------------
2024-08-13 16:04:44.692670-04:00Version:  2.11.3+dbb0883
2024-08-13 16:04:44.692683-04:00Date:     Tue Aug 13 16:04:44 EDT 2024
2024-08-13 16:04:44.692697-04:00User:     root
2024-08-13 16:04:44.692722-04:00PUID:     0
2024-08-13 16:04:44.692736-04:00User ID:  0
2024-08-13 16:04:44.692750-04:00PGID:     0
2024-08-13 16:04:44.692764-04:00Group ID: 0
2024-08-13 16:04:44.692777-04:00-------------------------------------
2024-08-13 16:04:44.692791-04:002024-08-13T16:04:44.692791090-04:00
2024-08-13 16:04:44.745762-04:00nginx: [warn] protocol options redefined for [::]:443 in /data/nginx/proxy_host/2.conf:19
2024-08-13 16:04:44.752853-04:00nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:44.754179-04:00nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:44.755425-04:00nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:44.756573-04:00nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:45.016534-04:00[Global   ] › ℹ  info      Using Sqlite: /data/etc/npm/database.sqlite
2024-08-13 16:04:46.073196-04:00[Migrate  ] › ℹ  info      Current database version: none
2024-08-13 16:04:46.112052-04:00[Certbot  ] › ▶  start     Installing duckdns...
2024-08-13 16:04:48.942292-04:00[Certbot  ] › ☒  complete  Installed duckdns
2024-08-13 16:04:48.943731-04:00[Setup    ] › ℹ  info      Added Certbot plugins duckdns
2024-08-13 16:04:48.944155-04:00[IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
2024-08-13 16:04:48.944481-04:00[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
2024-08-13 16:04:49.102026-04:00[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
2024-08-13 16:04:49.191230-04:00[SSL      ] › ℹ  info      Certbot Renewal Timer initialized
2024-08-13 16:04:49.191486-04:00[IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
2024-08-13 16:04:49.196281-04:00[Global   ] › ℹ  info      Backend PID 594 listening on port 48693 ...
2024-08-13 16:04:49.284639-04:00[Nginx    ] › ℹ  info      Starting Nginx
2024-08-13 16:04:49.319083-04:002024/08/13 16:04:49 [warn] 614#614: protocol options redefined for [::]:443 in /data/nginx/proxy_host/2.conf:19
2024-08-13 16:04:49.328938-04:002024/08/13 16:04:49 [warn] 614#614: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:49.331375-04:002024/08/13 16:04:49 [warn] 614#614: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:49.333336-04:002024/08/13 16:04:49 [warn] 614#614: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:49.335253-04:002024/08/13 16:04:49 [warn] 614#614: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2024-08-13 16:04:49.345895-04:002024/08/13 16:04:49 [notice] 614#614: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)
```

**Comment by Zoey2936** [comment]
Do you use any adcanced configs?

**Comment by cdcaish** [comment]
Don't think so.  All I did was create the proxy host and cert from the gui.

**Comment by Zoey2936** [comment]
can you send me the 2.conf file?

**Comment by cdcaish** [comment]
I decided to do some further testing and found that I had no issues running this in Docker on Windows OS.  This isn't the first time I've ran into quirks with TrueNAS and there docker implementation within kubernetes.  So I set up a test system with a nightly build of TrueNAS Electric Eel, which ditches kubernetes for a more native true docker/compose implementation.  I had no issues runing npmplus in Electric Eel.

I'm fine with writing this off as an issue with TrueNAS if you are.  I can run this in Docker on Windows until Electric Eel comes out. 
 Or if you want to continue troubleshooting that is fine too.
